src/pages/commits/ActiveCommits/buildVariantCard/FailedTaskStatusIcon.tsx

```diff
@@ -0,0 +1,84 @@
+import { useLazyQuery } from "@apollo/client";
+import styled from "@emotion/styled";
+import IconButton from "@leafygreen-ui/icon-button";
+import Tooltip from "@leafygreen-ui/tooltip";
+import { Body } from "@leafygreen-ui/typography";
+import { Skeleton } from "antd";
+import { Link } from "react-router-dom";
+import { TaskStatusIcon } from "components/TaskStatusIcon";
+import { getTaskRoute } from "constants/routes";
+import {
+  GetFailedTaskStatusIconTooltipQuery,
+  GetFailedTaskStatusIconTooltipQueryVariables,
+} from "gql/generated/types";
+import { GET_FAILED_TASK_STATUS_ICON_TOOLTIP } from "gql/queries";
+import { msToDuration } from "utils/string";
+
+interface FailedStatusIconProps {
+  taskId: string;
+  status: string;
+}
+export const FailedTaskStatusIcon: React.FC<FailedStatusIconProps> = ({
```

Comment:
This component is used to render normal tasks as well as failed tasks 
We only group tasks if there are no task filters. If there are filters applied we show the task icons individually. In that case we render normal tasks with non failure statuses with these icons. 
These still need tooltips but will not have any failures and no failed test results.
 https://github.com/evergreen-ci/spruce/blob/500c4bcb98110bac123a29d17a6af04eff22f0a8/src/pages/commits/ActiveCommits/BuildVariantCard.tsx#L31-L57
![image](https://user-images.githubusercontent.com/4605522/137986342-37c7f42e-adfe-4621-9a57-ce4d63085bb0.png)

Reply:
We should refactor this component to be generic for all tests and it should only fetch test results if the task has a failed status. Otherwise we should still show the tooltip.